### PR TITLE
[Workflows] Allow officers from all districts for US_ND

### LINKED ISSUE #2213: [US_ND] [Workflows] Disable district-level officer filter for entire state
If a user is marked as having a caseload and being assigned to a district, we currently restrict their list of officers to the other officers in their district.

This feature is not desired in ND.

We should add a flag in the `TenantConfig` objects to control whether this district-level filter is enabled on a state-by-state basis. Feel free to propose other ways to configure this.

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

This adds a flag to US_ND tenant `workflowsEnableAllDistricts` which will enable officers from all districts in the officer dropdown and caseloads lists. 

Another idea I had was to add a  `workflowsConfiguration` property to tenant config that's an array of strings like ["allowAllDistricts"], in the case that we think workflows will have more of these very specific configurations. 

The way it's currently set up means any new tenant will automatically be restricted until `workflowsEnableAllDistricts` is added to the config.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #2213

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/RootStore/__tests__/TenantStore.test.ts [hunk 1]
```diff
@@ -119,6 +130,51 @@ describe("TenantStore", () => {
...
+      jest.clearAllMocks();
+      jest.resetAllMocks();
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.resetAllMocks();
+    });
+
+    it("returns true when a current tenant enables viewing all districts", () => {
```
Comment (daschi) [inline src/RootStore/__tests__/TenantStore.test.ts:144]: I'm kinda stuck trying to figure out why this test passes when run alone but fails as a unit. It prints out US_MO as the current tenant ID, even though it should use the new mock rootStore? @macfarlandian @jovergaag have either of you dealt with this before?

Comment (macfarlandian) [reply src/RootStore/__tests__/TenantStore.test.ts:144]: weird ... usually that would be a case of state leakage between tests (e.g. a jest mock not being reset) but I don't see where that could be happening here. I'm stumped!

**Comment by Catacola** [comment]
This looks good! I like the idea of a list of options we could explicitly enable, but let's wait until we have 2 or 3 of them before building something more complex than just a boolean like this.

And I'm with Ian on the error.

File: src/RootStore/__tests__/TenantStore.test.ts (hunk 1 above)
Comment (macfarlandian) [reply src/RootStore/__tests__/TenantStore.test.ts:144]: was digging some more this morning and noticed that if I run just the local describe block this test passes and the second one fails. I think possibly the state is leaking via session storage which the tenant store reads from [here](https://github.com/Recidiviz/pulse-dashboard/blob/daschi-2213-district-filter/src/RootStore/TenantStore/TenantStore.ts#L36) so probably you need to clear it between tests?

Comment (daschi) [reply src/RootStore/__tests__/TenantStore.test.ts:144]: I found it! I needed to clear session storage

Comment (daschi) [reply src/RootStore/__tests__/TenantStore.test.ts:145]: Oh nice! I didn't see your comment before posting mine 😆

Comment (macfarlandian) [reply src/RootStore/__tests__/TenantStore.test.ts:145]: great minds

**Review by macfarlandian — APPROVED** [review]

**Pull request merged**